blender/arm/logicnode/arm_nodes.py

```diff
@@ -58,6 +89,82 @@ def add_output(self, socket_type: str, socket_name: str, default_value: Any = No
         return socket
 
 
+class NodeReplacement:
+    """
+    Represents a simple replacement rule, this can replace nodes of one type to nodes of a second type.
+    However, it is fairly limited. For instance, it assumes there are no changes in the type of the inputs or outputs
+    Second, it also assumes that node properties (especially EnumProperties) keep the same possible values.
+
+    - from_node, from_node_version: the type of node to be removed, and its version number
+    - to_node, to_node_version: the type of node which takes from_node's place, and its version number
+    - *_socket_mapping: a map which defines how the sockets of the old node shall be connected to the new node
+      {1: 2} means that anything connected to the socket with index 1 on the original node will be connected to the socket with index 2 on the new node
+    - property_mapping: the mapping used to transfer the values of the old node's properties to the new node's properties.
+      {"property0": "property1"} mean that the value of the new node's property1 should be the old node's property0's value.
+    - input_defaults: a mapping used to give default values to the inputs which aren't overridden otherwise.
+    - property_defaults: a mapping used to define the value of the new node's properties, when they aren't overridden otherwise.
+    """
+
+    def __init__(self, from_node: str, from_node_version: int, to_node: str, to_node_version: int,
+                 in_socket_mapping: Dict[int, int], out_socket_mapping: Dict[int, int], property_mapping: Dict[str, str] = {},
+                 input_defaults: Dict[int, any] = {}, property_defaults: Dict[str, any]={}):
```

Comment:
The way default arguments are used here is pretty dangerous: https://dev.to/sharmapacific/mutable-default-arguments-in-python-22ce.

Reply:
Dangit, I forgot about that.
I'm going to fix that (and the next comment).
Is there a way to add changes form this issue directly, or should I just amend the last commit/add a new commit?

Reply:
Just add a new commit, it will be automatically added to this PR

Reply:
aaand done.